### DISCUSSION TITLE
Extracting CLIP embeddings faster

### DISCUSSION THREAD

**Discussion by rxjx** [autogluon/autogluon#2172] [Q&A] [answered]
I'm trying to extract CLIP embeddings as described in [CLIP in AutoMM](https://auto.gluon.ai/dev/tutorials/multimodal/clip_embedding.html) for large batches of images/texts, say ~5000. For now I have to do this on a CPU but I'd like to speed it up as much as possible. What are the settings that ought to affect this? Would it be env.eval_batch_size_ratio or is that only for GPUs? Maybe env.num_workers to batch the number of images/texts being embedded?

**Comment by sxjscience** [comment, marked as answer]
Hi @rxjx , the `env.eval_batch_size_ratio` is also available in CPU.

**Reply by rxjx** [reply]
Thanks, I'll try it out.

**Reply by rxjx** [reply]
Does that mean that "env.per_gpu_batch_size" and "env.num_gpus" will also apply to CPUs?

**Reply by rxjx** [reply]
Looks like env.eval_batch_size_ratio doesn't help.

**Reply by sxjscience** [reply]
Yeah, this is really related how to accelerate the inference in CPU.

We have a PR that is converting the model to ONNX: https://github.com/awslabs/autogluon/pull/2149. Usually, ONNX runtime will be faster than native pytorch.
